Fail fast and point to docs for 'skaffold init' on helm projects (#4396)
GoogleContainerTools/skaffold: pkg/skaffold/initializer/testdata/init/helm-deployment/README.md

### Example: deploy multiple releases with Helm

> You can deploy multiple releases with skaffold, each will need a chartPath, a values file, and namespace.
> Skaffold can inject intermediate build tags in the the values map in the skaffold.yaml.

Reviewer: ```suggestion
You can deploy multiple releases with Skaffold, each will need a chartPath, a values file, and an optional namespace.
Skaffold can inject intermediate build tags in the the values map in the `skaffold.yaml`.
```

Let's walk through the skaffold yaml:

> We'll be building an image called `skaffold-helm`, and it's a dockerfile, so we'll add it to the artifacts.

Reviewer: Keep the tenses the same as later.
```suggestion
We are building an image called `skaffold-helm`, and it builds with Docker, so we add it to the artifacts.
```

Author: actually we shouldn't be updating this - this was copied directly from https://github.com/GoogleContainerTools/skaffold/blob/master/examples/helm-deployment/README.md :) we can update that later but i'll leave this as is in this PR

```yaml
build:
  artifacts:
  - image: skaffold-helm
```

Now, we want to deploy this image with helm.
We add a new release in the helm part of the deploy stanza.

```yaml
deploy:
  helm:
    releases:
    - name: skaffold-helm
      chartPath: charts
      # namespace: skaffold
      artifactOverrides:
        image: skaffold-helm
      valuesFiles:
      - values.yaml
```

> This part tells Skaffold to set the `image` parameter of the values file to the built `skaffold-helm` image and tag.

Reviewer: Skaffold will add an `image` parameter if not found, right?
